arbiter-core/src/tests/middleware_instructions.rs

```diff
@@ -300,6 +303,117 @@ async fn set_gas_price() {
     assert_eq!(client.get_gas_price().await.unwrap(), test_gas_price);
 }
 
+#[tokio::test]
+async fn get_transaction_count() {
+    let (_environment, client) = startup_user_controlled().unwrap();
+    let nonce = client
+        .get_transaction_count(client.address(), Some(0.into()))
+        .await
+        .unwrap();
+    assert_eq!(nonce.as_u64(), 0_u64);
+
+    let _arbiter_token = deploy_arbx(client.clone()).await.unwrap();
+    let nonce_after_tx = client
+        .get_transaction_count(client.address(), Some(0.into()))
+        .await
+        .unwrap();
+    assert_eq!(nonce_after_tx.as_u64(), 1_u64);
+}
+
+#[tokio::test]
+async fn create_nonce_middleware() {
+    let (_manager, client) = startup_user_controlled().unwrap();
```

Comment:
All these spots you have `_manager` are not correct. The startup functions return `(Environment, Arc<RevmMiddleware>)`